### PR TITLE
Fix a few integration tests that failed often

### PR DESCRIPTION
**Pull request by ankur22**
## What?

Fixes a few of the integration tests that were failing often locally (not related to CI flakey tests, but it might help there too).

## Why?

Better DX.

## Checklist

- [x] I have performed a self-review of my code
- [ ] I have added tests for my changes
- [ ] I have commented on my code, particularly in hard-to-understand areas

### CODE REVIEW

File: tests/browser_test.go [hunk 1]
```diff
@@ -99,8 +99,19 @@ func TestTmpDirCleanup(t *testing.T) {
 
 	b.Close()
 
-	matches, err = filepath.Glob(tmpDirPath + "/xk6-browser-data-*")
-	assert.NoError(t, err)
+	// We need to wait for something (k6 browser, chromium or the os) to
+	// actually complete the removal of the directory. It's a race condition.
+	// To try to mitigate the issue, we're adding a retry which waits half a
+	// second if the dir still exits.
```
Comment (inancgumus) [inline tests/browser_test.go:105]: Nice comment. I'd also add `TODO` note here, saying, fix this race condition - or might even make sense to create an issue for it.

**Review by inancgumus — APPROVED** [review]
Thanks!

File: k6ext/k6test/vu.go [hunk 2]
```diff
@@ -139,7 +139,10 @@ func (v *VU) RunOnEventLoop(tb testing.TB, js string, args ...any) (sobek.Value,
 func (v *VU) RunAsync(tb testing.TB, js string, args ...any) (sobek.Value, error) {
 	tb.Helper()
 
-	return v.RunOnEventLoop(tb, fmt.Sprintf("(async function() { "+js+" })();", args...))
```
Comment (ankur22) [inline k6ext/k6test/vu.go:142]: Hoping to fix https://github.com/grafana/xk6-browser/actions/runs/10773846755/job/29877212064 with this change 🤞

Comment (inancgumus) [reply k6ext/k6test/vu.go:142]: The number of args don't match to the number of values, I guess.

Comment (ankur22) [reply k6ext/k6test/vu.go:142]: Never seen this issue before 🤷 Seems to only occur with the tip.

Comment (inancgumus) [reply k6ext/k6test/vu.go:142]: Related (IDK :))?

* https://github.com/argoproj/argo-cd/issues/19772
* https://cs.opensource.google/go/x/tools/+/refs/tags/v0.24.0:go/analysis/passes/printf/printf.go;l=502-534
* https://code.waarp.fr/apps/gateway/gateway/-/blob/master/.golangci.yml#L841-844

Comment (ankur22) [reply k6ext/k6test/vu.go:142]: Right, i get it, it's a new change in tip, that's why it's failing on the build step 👍

Comment (ankur22) [reply k6ext/k6test/vu.go:142]: I still don't understand why the original change wasn't building, but the new change seems to have solved it.

**Review by inancgumus — APPROVED** [review]

**Pull request merged**